src/main/java/com/maxmind/db/Networks.java

```diff
@@ -0,0 +1,184 @@
+     * @throws ClosedDatabaseException Exception for a closed database.
+     */
+    Networks(Reader reader, boolean skipAliasedNetworks) 
+        throws ClosedDatabaseException {
+        this(reader, skipAliasedNetworks, new NetworkNode[]{});
+    }
+
+    /**
+     * Constructs a Networks instance.
+     * @param reader The reader object.
+     * @param skipAliasedNetworks The boolean to skip aliased networks.
+     * @param nodes The initial nodes array to start Networks iterator with.
+     * @throws ClosedDatabaseException Exception for a closed database.
+     */
+    Networks(Reader reader, boolean skipAliasedNetworks, NetworkNode[] nodes) 
+        throws ClosedDatabaseException {
+        this.reader = reader;
+        this.skipAliasedNetworks = skipAliasedNetworks;
+        this.buffer = reader.getBufferHolder().get();
+        this.nodes = new Stack<NetworkNode>();
+        for (NetworkNode node : nodes) {
+            this.nodes.push(node);
+        }
+    }
+
+    /**
+     * Constructs a Networks instance with skipAliasedNetworks set to false by default.
+     * @param reader The reader object.
+     */
+    Networks(Reader reader) throws ClosedDatabaseException {
+        this(reader, false);
+    }
+
+    /**
+     * Sets the Class for the data type in DataRecord.
+     * @param cls The class object. ( For example, Map.class )
+     */
+    public void setDataClass(Class<T> cls) {
+        this.typeParameterClass = cls;
+    }
+
+    /**
+     * Returns the next DataRecord. You need to set the class using
+     * prepareForClass before calling next. 
+     * For example,
+     *  networks.prepareForClass(Map.Class);
+     *  Map test = networks.next();
+     */
+    @Override
+    public DatabaseRecord<T> next() {
+        try {
+            T data = this.reader.resolveDataPointer(
+                this.buffer, this.lastNode.pointer, this.typeParameterClass);
+
+            byte[] ip = this.lastNode.ip;
+            int prefixLength = this.lastNode.prefix;
+
+            // We do this because uses of SkipAliasedNetworks expect the IPv4 networks
+            // to be returned as IPv4 networks. If we are not skipping aliased
+            // networks, then the user will get IPv4 networks from the ::FFFF:0:0/96
+            // network.
+            if (this.skipAliasedNetworks && isInIpv4Subtree(ip)) {
+                ip = Arrays.copyOfRange(ip, 12, ip.length);
+                prefixLength -= 96;
+            }
+
+            // If the ip is in ipv6 form, drop the prefix manually
+            // as InetAddress converts it to ipv4.
+            InetAddress ipAddr = InetAddress.getByAddress(ip);
+            if (ipAddr instanceof Inet4Address && ip.length > 4 && prefixLength > 96) {
+                prefixLength -= 96;
+            }
+
+            return new DatabaseRecord<T>(data, InetAddress.getByAddress(ip), prefixLength);
+        } catch (IOException e) {
+            throw new NetworksIterationException(e.getMessage());
```

Comment:
I think we should wrap the whole and set the `cause` to the original exception. This will allow users to access all the data in the original.

```suggestion
            throw new NetworksIterationException(e);
```